goldmane/pkg/aggregator/bucketing/bucket_ring.go

```diff
@@ -274,15 +274,45 @@ func (r *BucketRing) nowIndex() int {
 	return r.indexSubtract(r.headIndex, 1)
 }
 
-// FlowCollection returns a collection of flows to emit, or nil if we are still waiting for more data.
-// The BucketRing builds a FlowCollection by aggregating flow data from across a window of buckets. The window
-// is a fixed size (i.e., a fixed number of buckets), and starts a fixed period of time in the past in order to allow
-// for statistics to settle down before publishing.
-func (r *BucketRing) FlowCollection() *FlowCollection {
+func (r *BucketRing) FlowCollections() []*FlowCollection {
 	// Determine the newest bucket in the aggregation - this is always pushAfter buckets back from "now".
 	endIndex := r.indexSubtract(r.nowIndex(), r.pushAfter)
 	startIndex := r.indexSubtract(endIndex, r.bucketsToAggregate)
 
+	// maxCollections defines the maximum number of collections to build for any given call. Each collection is
+	// 5 minutes long, so 12 collections will cover a full hour.
+	maxCollections := 12
```

Comment:
What's the reason for needing a maximum number here?

Also, isn't that 5 minute window configurable (so should this be based off a configurable window)?

Reply:
Yeah, it probably should be adjusted to at least be based on the configurable window.

The original reason I added this was to avoid looping forever (since it's a ring!) 

Other ways to do this would just be to break when we loop around one complete ring, which is probably what we want.